corehq/apps/locations/resources/v0_6.py

```diff
@@ -44,61 +51,78 @@ def dehydrate(self, bundle):
 
     def obj_create(self, bundle, **kwargs):
         domain = kwargs['domain']
-        if 'name' not in bundle.data:
-            raise BadRequest("'name' is a required field.")
-        if SQLLocation.objects.filter(domain=domain, site_code=bundle.data['site_code']).exists():
-            raise BadRequest("Location on domain with site code already exists.")
+        if 'name' not in bundle.data or 'location_type_code' not in bundle.data:
+            raise BadRequest("'name' and 'location_type_code' are required fields.")
```

Comment:
Ah that makes sense that tastypie would look at the model definition.  Kinda surprised `null=True` for `name` though, since we definitely expect it to be present, _and_ because it's a charfield (where `null=True` is generally frowned upon, as `''` should be used instead)